### PR TITLE
frontend: add build metadata

### PR DESCRIPTION
**Pull request by mwasilew**
Build metadata is composed of all test runs metadata

Signed-off-by: Milosz Wasilewski <milosz.wasilewski@linaro.org>

### CODE REVIEW

File: squad/frontend/views.py [hunk 1]
```diff
@@ -58,9 +58,13 @@ def build(request, group_slug, project_slug, version):
     group = Group.objects.get(slug=group_slug)
     project = group.projects.get(slug=project_slug)
     build = project.builds.prefetch_related('test_runs', 'test_runs__status', 'test_runs__status__suite', 'test_runs__status__test_run__environment').get(version=version)
+    metadata_dict = {}
+    for testrun in build.test_runs.all():
+        metadata_dict.update(json.loads(testrun.metadata_file))
```
Comment (terceiro) [inline squad/frontend/views.py]: do we want to include *all* metadata entries in the build metadata? I assume several of the items will be specific to each test run?

maybe the build metadata should be the _intersection_ of the test run metadata?

Comment (terceiro) [inline squad/frontend/views.py]: or at least the entries that are not different across the test runs

Comment (mwasilew) [inline squad/frontend/views.py]: Sounds like a good idea. I'll fix it.

File: squad/frontend/views.py [hunk 2]
```diff
@@ -58,9 +58,23 @@ def build(request, group_slug, project_slug, version):
...
+        testrun_metadata = json.loads(testrun.metadata_file)
+        if not metadata_dict:
+            metadata_dict.update(testrun_metadata)
+        else:
+            for key, value in testrun_metadata.items():
+                if key in metadata_dict.keys() and metadata_dict[key] != testrun_metadata[key]:
+                    del metadata_dict[key]
+            for key in metadata_dict.keys() & testrun_metadata.keys():
+                if key not in metadata_dict.keys():
+                    del metadata_dict[key]
```
Comment (terceiro) [inline squad/frontend/views.py]: there is a way simpler way of intersecting two dicts:

```python
>>> x = {1:2, 3:4}
>>> y = {1:2, 5:6}
>>> dict(x.items() & y.items())
{1: 2}
```

**Comment by mwasilew** [comment]
Fixed. I think this is the simplest way of doing things.

**Pull request merged**